packages/react/src/createReactQueryHooks.tsx

```diff
@@ -60,6 +61,31 @@ interface UseTRPCMutationOptions<TInput, TError, TOutput>
   extends UseMutationOptions<TOutput, TError, TInput>,
     TRPCUseQueryBaseOptions {}
 
+type Override<A, B> = {
+  [K in keyof A]: K extends keyof B ? B[K] : A[K];
+};
+type Equals<X, Y> = (<T>() => T extends X ? 1 : 2) extends <T>() => T extends Y
+  ? 1
+  : 2
+  ? true
+  : false;
```

Comment:
`Override`, and `Equals` probably shouldn't live in `createReactQueryHooks`. not sure where to move them tho